src/globus_sdk/services/search/client.py

```diff
@@ -278,17 +279,13 @@ def search(
 
                 .. expandtestfixture:: search.search
         """  # noqa: E501
-        if query_params is None:
-            query_params = {}
-        query_params.update(
-            {
-                "q": q,
-                "offset": offset,
-                "limit": limit,
-                "advanced": advanced,
-            }
-        )
-
+        query_params = {
+            **(query_params or {}),
+            "q": q,
+            "offset": offset,
+            "limit": limit,
+            "advanced": advanced,
+        }
```

Comment:
I think that this should be reversed so that users can put all of their parameters into `query_params` without them getting overwritten.

```suggestion
        query_params = {
            "q": q,
            "offset": offset,
            "limit": limit,
            "advanced": advanced,
            **(query_params or {}),
        }
```

I'm thinking of this use case:

```python
fn(query_params={"offset": 33, "limit": 77})  # called with offset=0, limit=10
```

This feedback may apply to other areas of the code, in the specific case that we're providing default values that could overwrite a user's `query_params` values.

Reply:
I completely agree but I want to call out that such a change would be a different behavior from the one currently in the SDK. 

Currently `query_params` is constructed like so:
```
if query_params is None:
    query_params = {}
query_params.update(
    {
        "q": q,
        "offset": offset,
        "limit": limit,
        "advanced": advanced,
    }
)
```
which I believe is doing what you mention above of overriding the query param fields with the method defaults. I think you are correct that it is the wrong behavior, but I wanna explicitly call out that it would be different before going ahead with your proposed change. Thoughts?

Reply:
Part of the resolution from discussion is that the defaults should be `MISSING` for these values -- since there is a service-side default. I missed that when commenting before, but that's a good improvement to add and it resolves this case.

I'm not sure why the SDK has defaults for these; as the original author here, I can't recall / speak to what justification I might have had for doing that.